torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -733,6 +742,12 @@ class TORCH_API ProcessGroupNCCL : public Backend {
       const std::chrono::time_point<std::chrono::steady_clock>& wakeUpTime,
       size_t timeout_sec = 30);
 
+  // A helper function to wait for a future to complete or timeout.
+  void waitForFutureOrTimeout(
+      std::future<bool>& fut,
+      const std::chrono::milliseconds& timeOutMilSec,
+      const std::string& futDescription);
+
```

Comment:
nit: I wonder if this function needs to be a method of `ProcessGroupNCCL`? The declaration and implementation seem quite generic.

Reply:
The only thing special to ProcessGroupNCCL is the exception info including the logPrefix making things easier to debug. with c++ 20 with coroutines, we should not need this helper.